Documentation/Global/Form/Password.rst

```diff
@@ -3,23 +3,102 @@
 ..  include:: /Includes.rst.txt
 ..  _typo3-fluid-form-password:
 
-============================================
+=============================================
 Form.password ViewHelper `<f:form.password>`
-============================================
+=============================================
 
 ..  typo3:viewhelper:: form.password
-    :source: ../../Global.json
-    :display: tags,description,gitHubLink,arguments
+    :source: /Global.json
+    :display: tags,description,gitHubLink
+    :noindex:
 
-..  _typo3-fluid-form-password-example:
+..  warning::
+    Password fields should never be prefilled with sensitive data.
+    **Always** use secure handling on the server side to hash and verify passwords.
+    **Never** persist or display passwords as plain text. See also
+    `Excursion: Validation and hashing of password fields <https://docs.typo3.org/permalink/t3viewhelper:typo3-fluid-form-password-example-validation>`_.
 
-Examples
-========
+..  contents:: Table of contents
 
-Example::
+..  _typo3-fluid-form-password-example-basic:
 
-   <f:form.password name="myPassword" />
+A basic password field
+======================
 
-Output::
+The simplest usage sets a name and renders a password input:
 
-   <input type="password" name="myPassword" value="" />
+..  tabs::
+
+    ..  group-tab:: Fluid
+
+        ..  literalinclude:: _codesnippets/_PasswordFormBasic.html
+            :caption: Resources/Private/Templates/User/LoginForm.html
+
+    ..  group-tab:: Output
+
+        ..  code-block:: html
+
+            <input type="password" name="myPassword" value="" />
+
+..  _typo3-fluid-form-password-example-model:
+
+Binding to a model property
+===========================
+
+You can bind the password field to a property in an Extbase model.
+This requires the form to define the `object` and `objectName` as well.
+
+..  literalinclude:: _codesnippets/_PasswordFormBound.html
+    :caption: Resources/Private/Templates/User/RegisterForm.html
+
+..  _typo3-fluid-form-password-example-validation:
+
+Excursion: Validation and hashing of password fields
+====================================================
+
+A password should be validated against the
+`Password policy <https://docs.typo3.org/permalink/t3coreapi:password-policies>`_
+and **must** always be hashed before saving. **Never** save a plain text
+password to the database.
+
+..  tabs::
+
+    ..  group-tab:: Model
+
+        ..  literalinclude:: _codesnippets/_UserModel.php
+            :caption: Classes/Domain/Model/User.php
+
+    ..  group-tab:: Controller
+
+        ..  literalinclude:: _codesnippets/_UserController.php
+            :caption: Classes/Controller/UserController.php
+
```

Comment:
How about something like the following, because I think inlining validation to actions is not the best practice we should enforce, but I understand you chose this for brevity.

The example above shows validation inlined in an Action of the controller for brevity.
For re-usability and better architecture following best practices, you can create a custom validator for this (see: custom validators).

Reply:
done